src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -176,6 +178,13 @@ public void onRunFinish(boolean succeeded, RuntimeContext context) {
     }
   }
 
+  @Override
+  public SparkPushDataset<StructuredRecord> getSparkPushProvider(SQLPushRequest sqlPushRequest)
```

Comment:
This ties SQL Engine to spark. Can be fine for POC, but we would need more structure similar to plugin approach for merging. If we look into running full ELT in future we should not tie to Spark.